modules/evm-accounts/src/lib.rs

```diff
@@ -253,6 +254,15 @@ impl<T: Trait> AddressMapping<AccountId32> for EvmAddressMapping<T> {
 	}
 }
 
+pub struct EvmAccountMapping<T>(sp_std::marker::PhantomData<T>);
+impl<T: Trait> AccountMapping<AccountId32> for EvmAccountMapping<T> {
```

Comment:
can you do `where T::AccountId: From<AccountId32>` to avoid the encode / decode hack?